docs/library/forms/switch.md

```diff
@@ -0,0 +1,216 @@
+---
+components:
+    - rx.radix.themes.Switch
+---
+
+```python exec
+import reflex as rx
+from reflex.components.radix.themes.components import *
+from reflex.components.radix.themes.layout import *
+from reflex.components.radix.themes.typography import *
+from pcweb.templates.docpage import style_grid
+```
+
+# Switch
+
+
+A toggle switch alternative to the checkbox.
+
+## Basic Example
+
+```python demo
+text(
+    flex(
+        switch(default_checked=True),
+        "Sync Settings",
+        gap="2",
+    )
+)
+
+```
+
+## Usage
+
+
+### Submitting a form using switch
+
+The `name` of the switch is needed to submit with its owning form as part of a name/value pair.
+
+When the `required` prop is `True`, it indicates that the user must check the switch before the owning form can be submitted.
+
+The `value` prop is only used for form submission, use the `checked` prop to control state of the `switch`.
+
+```python demo exec
+class FormSwitchState(rx.State):
+    form_data: dict = {}
+
+    def handle_submit(self, form_data: dict):
+        """Handle the form submit."""
+        self.form_data = form_data
+
+
+def form_switch():
+    return rx.vstack(
+        rx.form(
+            rx.vstack(
+                switch(name="s1"),
+                switch(name="s2"),
+                switch(name="s3", required=True),
+                rx.button("Submit", type_="submit"),
+                width="100%",
+            ),
+            on_submit=FormSwitchState.handle_submit,
+            reset_on_submit=True,
+            width="100%",
+        ),
+        rx.divider(),
+        rx.heading("Results"),
+        rx.text(FormSwitchState.form_data.to_string()),
+        width="100%",
+    )
+```
+
+
+
+### High control of switch component
+
+The `checked` prop is used to control the state of the switch. 
+
+The event handler `on_checked_change` is called when the state of the switch changes.
```

Comment:
Technically the event handler is called `change_checked` below, the "event" is `on_checked_change`

Reply:
changed